sway-core/src/lib.rs

```diff
@@ -222,7 +222,7 @@ fn module_path(parent_module_dir: &Path, dep: &sway_ast::Dependency) -> PathBuf
 pub struct CompiledAsm(pub FinalizedAsm);
 
 /// The bytecode for a sway program.
-pub struct CompiledBytecode(pub Vec<u8>);
+pub struct CompiledBytecode(pub (Vec<u8>, BTreeMap<String, u64>));
```

Comment:
Could be nice to have a type alias for this for readability, e.g.

```rust
pub type ConfigConstOffsets = BTreeMap<String, u64>;
```
Or alternatively add field names
```rust
pub struct CompiledBytecode {
    pub bytecode: Vec<u8>,
    pub config_const_offsets: BTreeMap<String, u64>,
}
```

Reply:
Great idea! I went with the second solution and also moved the same logic to `finalized_asm.rs`.